api-server/src/server/boot/user.js

```diff
@@ -129,7 +140,7 @@ function createPostMsUsername(app) {
       });
     }
 
-    const msTranscriptId = msTranscriptUrl.split('/').pop();
+    const msTranscriptId = getMsTranscriptApiUrl(msTranscriptUrl);
     const msTranscriptApiUrl = `https://learn.microsoft.com/api/profiles/transcript/share/${msTranscriptId}`;
```

Comment:
```suggestion
    const msTranscriptApiUrl = getMsTranscriptApiUrl(msTranscriptUrl);
```

The new function is returning the whole URL, not the transcript ID. So we were making requests to `https://learn.microsoft.com/api/profiles/transcript/share/https://learn.microsoft.com/api/profiles/transcript/share/abcdefghijklmno` (not correct)

Reply:
I should have written tests :face_exhaling: